mwdb/web/src/components/Navigation.js

```diff
@@ -112,6 +112,27 @@ function RemoteDropdown() {
     );
 }
 
+function UploadButton() {
+    const auth = useContext(AuthContext);
+    const buttonLink = auth.hasCapability("adding_files") ? (
+        <Link className="nav-link" to={"/upload"}>
+            <FontAwesomeIcon className="navbar-icon" icon={faUpload} />
+            Upload
+        </Link>
+    ) : (
```

Comment:
I wonder if it is better to hide this button completely when adding_files permission is missing.

Reply:
Hmm, I was thinking what option is the least confusing and decided to make an exception for Upload. I think if you just hide the Upload, it may trigger some questions like "Where is Upload button?" because it's one of the most basic functions in MWDB. If you show the disabled button and tooltip, it's clear that you're in "read-only" mode.